### DISCUSSION TITLE
Problema network con HA Supervised

### DISCUSSION THREAD

**Discussion by SuperSpe** [indomus/forum#2753] [HUB: Home Assistant]
Buongiorno, è la prima volta che installo HA Supervised. Smanetto abbastanza sia con HA che con Docker, ma questa non riesco a risolverla.
Praticamente all'avvio del raspberry, l'host esce in internet correttamente. Nel momento in cui partono i container di docker relativi ad HA, l'host perde la connessione internet. Ho indagato ed "colpa" delle route statiche che aggiunge docker con le sue interfacce virtuali, ma mi sembra ci sia troppa roba. Vorrei capire come configurare il servizio di docker in modo che usi il gateway corretto, se questa è la strada.
Queste le route prima dell'avvio di docker
```
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
default         fritz.box       0.0.0.0         UG    0      0        0 eth0
default         fritz.box       0.0.0.0         UG    202    0        0 eth0
172.17.0.0      0.0.0.0         255.255.0.0     U     0      0        0 docker0
172.20.0.0      0.0.0.0         255.255.0.0     U     0      0        0 br-01369783317d
172.22.0.0      0.0.0.0         255.255.0.0     U     0      0        0 br-da80b4b7dc56
172.30.32.0     0.0.0.0         255.255.254.0   U     0      0        0 hassio
192.168.178.0   0.0.0.0         255.255.255.0   U     202    0        0 eth0
pi@raspberrypi:~ $
```
Queste le route dopo l'avvio di docker, la default viene sovrascritta dall'altra sull'interfaccia veth0bd0926
```
Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
0.0.0.0         0.0.0.0         255.255.255.255 UH    0      0        0 veth0bd0926
0.0.0.0         0.0.0.0         255.255.255.255 UH    0      0        0 vethd5a2491
0.0.0.0         0.0.0.0         255.255.255.255 UH    0      0        0 veth7f3c535
0.0.0.0         0.0.0.0         255.255.255.255 UH    0      0        0 vethc6d2155
0.0.0.0         0.0.0.0         255.255.255.255 UH    0      0        0 veth93a82f1
0.0.0.0         0.0.0.0         255.255.255.255 UH    0      0        0 veth2240900
0.0.0.0         0.0.0.0         255.255.255.255 UH    0      0        0 veth036ff18
0.0.0.0         0.0.0.0         255.255.255.255 UH    0      0        0 veth1ef62a9
0.0.0.0         0.0.0.0         255.255.255.255 UH    0      0        0 vetha15cb26
0.0.0.0         0.0.0.0         255.255.255.255 UH    0      0        0 vethedaaf37
default         0.0.0.0         0.0.0.0         U     0      0        0 veth0bd0926
default         fritz.box       0.0.0.0         UG    0      0        0 eth0
default         fritz.box       0.0.0.0         UG    100    0        0 eth0
default         fritz.box       0.0.0.0         UG    202    0        0 eth0
link-local      0.0.0.0         255.255.0.0     U     209    0        0 veth2240900
link-local      0.0.0.0         255.255.0.0     U     211    0        0 veth036ff18
link-local      0.0.0.0         255.255.0.0     U     213    0        0 vethedaaf37
link-local      0.0.0.0         255.255.0.0     U     215    0        0 veth1ef62a9
link-local      0.0.0.0         255.255.0.0     U     217    0        0 vetha15cb26
link-local      0.0.0.0         255.255.0.0     U     219    0        0 veth93a82f1
link-local      0.0.0.0         255.255.0.0     U     221    0        0 vethc6d2155
link-local      0.0.0.0         255.255.0.0     U     223    0        0 veth7f3c535
link-local      0.0.0.0         255.255.0.0     U     225    0        0 vethd5a2491
link-local      0.0.0.0         255.255.0.0     U     227    0        0 veth0bd0926
172.17.0.0      0.0.0.0         255.255.0.0     U     0      0        0 docker0
172.20.0.0      0.0.0.0         255.255.0.0     U     0      0        0 br-01369783317d
172.22.0.0      0.0.0.0         255.255.0.0     U     0      0        0 br-da80b4b7dc56
172.30.32.0     0.0.0.0         255.255.254.0   U     0      0        0 hassio
192.168.178.0   0.0.0.0         255.255.255.0   U     202    0        0 eth0

```

Questa la lista degli ip dei vari container
hassio_supervisor 172.17.0.3
hassio_observer	172.30.32.6
hassio_multicast NO IP
hassio_dns	172.30.32.3	
hassio_cli 172.30.32.5
hassio_audio 172.30.32.4

e le network di docker

NETWORK ID     NAME                                DRIVER    SCOPE
f86ae9c71adb   bridge                              bridge    local
da80b4b7dc56   docker-compose-lemp-stack_default   bridge    local
333bb4d4bd69   hassio                              bridge    local
4f81d8e98706   host                                host      local
aea56de2b1f5   none                                null      local
01369783317d   tautulli_default                    bridge    local

grazie

**Comment by KC-inDomus** [comment]
(a latere) la strada corretta, se sei uno che smanetta con HA e Docker, è NON usare Supervised, ma core@docker.

**Reply by SuperSpe** [reply]
Arrivo da HA OS quindi non mi sono mai posto il problema, ora ho provato una strada diversa ma sono partito sbagliato, non avevo capito bene la differenza tra supervised e core.
però questo è un problema di docker quindi non so neanche quanto centri HA

**Comment by SuperSpe** [comment]
Ho risolto. Il problema è di connman.
Modificato il file /etc/connman/main.conf e abilitato la seguente variabile che ignora le interfacce virtuali

[General]
NetworkInterfaceBlacklist=vmnet,vboxnet,virbr,ifb,docker,veth,eth,wlan1
